integrations/viem-v2/src/index.ts

```diff
@@ -0,0 +1,186 @@
+import { KeyFetcher, wrap } from "@oasisprotocol/sapphire-paratime";
+import {
+	type Client,
+	type SerializeTransactionFn,
+	type Transport,
+	type WalletClient,
+	custom,
+	defineChain,
+	serializeTransaction,
+} from "viem";
+
+declare let process: {
+	env: {
+		SAPPHIRE_LOCALNET_HTTP_PROXY_PORT?: string;
+	};
+};
+
+/**
+ * This environment variable allows for the sapphire-localnet port to be
+ * overridden via the command-line. This is useful for debugging with a proxy.
+ *
+ * Note: this will fail gracefully in-browser
```

Comment:
How about checking if `window` exists, and not even execute the code, that depends on Node.js runtime?

Reply:
I assumed some people may use a `node-process` polyfill